lib/active_merchant/billing/gateways/global_collect.rb

```diff
@@ -250,20 +255,49 @@ def add_payment(post, payment, options)
         month = format(payment.month, :two_digits)
         expirydate = "#{month}#{year}"
         pre_authorization = options[:pre_authorization] ? 'PRE_AUTHORIZATION' : 'FINAL_AUTHORIZATION'
-        post['cardPaymentMethodSpecificInput'] = {
+        specifics_inputs = {
           'paymentProductId' => BRAND_MAP[payment.brand],
           'skipAuthentication' => 'true', # refers to 3DSecure
           'skipFraudService' => 'true',
           'authorizationMode' => pre_authorization
         }
-        post['cardPaymentMethodSpecificInput']['requiresApproval'] = options[:requires_approval] unless options[:requires_approval].nil?
+        specifics_inputs['requiresApproval'] = options[:requires_approval] unless options[:requires_approval].nil?
+        if payment.is_a?(NetworkTokenizationCreditCard) || (payment.is_a?(CreditCard) && options[:google_pay_pan_only])
+          specifics_inputs['paymentProductId'] = options[:google_pay_pan_only] ? BRAND_MAP[:google_pay] : BRAND_MAP[payment.source]
+          post['mobilePaymentMethodSpecificInput'] = specifics_inputs
+          add_decrypted_payment_data(post, payment, options, expirydate)
+        elsif payment.is_a?(CreditCard)
+          post['cardPaymentMethodSpecificInput'] = specifics_inputs.merge({
+            'card' => {
+              'cvv' => payment.verification_value,
+              'cardNumber' => payment.number,
+              'expiryDate' => expirydate,
+              'cardholderName' => payment.name
+            }
+          })
```

Comment:
Not a huge deal, but I feel like this method is getting a little messy and difficult to reason about. I would suggest breaking it out into separate `add_credit_card` and `add_network_tokenization_credit_card` methods, making this method a cleaner conditional.